src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1950,6 +1961,25 @@ public bool Equals(ReadOnlySpan<char> span)
         /// are removed from this builder.
         /// </remarks>
         public StringBuilder Replace(string oldValue, string? newValue, int startIndex, int count)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(oldValue);
+            
+            ReadOnlySpan<char> newValueSpan = (newValue == null) ? ReadOnlySpan<char>.Empty : newValue.AsSpan();
+            return Replace(oldValue.AsSpan(), newValueSpan, startIndex, count);
```

Comment:
```suggestion
            return Replace(oldValue.AsSpan(), newValue.AsSpan(), startIndex, count);
```

Reply:
I was under the assumption that calling `.AsSpan()` on a `null` string would cause an exception, but it just returns an empty `ReadOnlySpan<char>` - thanks for the input.